### ISSUE TITLE
Test failures in master

### ISSUE THREAD

**Issue by asmeurer** [asmeurer/pytest-flakes#28]
It looks like there are test failures in master. I did a test Travis build on my fork at https://travis-ci.org/github/asmeurer/pytest-flakes. 

The test failure looks like

```
______________________________ test_syntax_error _______________________________
testdir = <Testdir local('/tmp/pytest-of-travis/pytest-0/test_syntax_error0')>
    def test_syntax_error(testdir):
        testdir.makeini("""
    [pytest]
    python_files=check_*.py
    """)
        testdir.makepyfile("""
    for x in []
        pass
    """)
        result = testdir.runpytest("--flakes", "--ignore", testdir)
>       assert "1: invalid syntax" in result.stdout.str()
E       AssertionError: assert '1: invalid syntax' in ''
E        +  where '' = <bound method LineMatcher.str of <_pytest.pytester.LineMatcher object at 0x7f3e69cf66a0>>()
E        +    where <bound method LineMatcher.str of <_pytest.pytester.LineMatcher object at 0x7f3e69cf66a0>> = <_pytest.pytester.LineMatcher object at 0x7f3e69cf66a0>.str
E        +      where <_pytest.pytester.LineMatcher object at 0x7f3e69cf66a0> = <RunResult ret=ExitCode.INTERNAL_ERROR len(stdout.lines)=0 len(stderr.lines)=40 duration=0.01s>.stdout
/home/travis/build/asmeurer/pytest-flakes/test_flakes.py:24: AssertionError
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/pytester.py", line 932, in runpytest_inprocess
    reprec = self.inline_run(*args, **kwargs)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/pytester.py", line 898, in inline_run
    ret = pytest.main(list(args), plugins=plugins)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/config/__init__.py", line 105, in main
    config = _prepareconfig(args, plugins)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/config/__init__.py", line 258, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/helpconfig.py", line 90, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/config/__init__.py", line 836, in pytest_cmdline_parse
    self.parse(args)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/config/__init__.py", line 1044, in parse
    self._preparse(args, addopts=addopts)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/config/__init__.py", line 980, in _preparse
    self._initini(args)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/config/__init__.py", line 903, in _initini
    args, namespace=copy.copy(self.option)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/config/argparsing.py", line 158, in parse_known_and_unknown_args
    return optparser.parse_known_args(strargs, namespace=namespace)
  File "/opt/python/3.6.7/lib/python3.6/argparse.py", line 1766, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/opt/python/3.6.7/lib/python3.6/argparse.py", line 1807, in _parse_known_args
    option_tuple = self._parse_optional(arg_string)
  File "/home/travis/build/asmeurer/pytest-flakes/.tox/py36/lib/python3.6/site-packages/_pytest/config/argparsing.py", line 427, in _parse_optional
    if not arg_string[0] in self.prefix_chars:
TypeError: 'Testdir' object does not support indexing
```

**Comment by asmeurer** [comment]
The other failure I believe is caused by https://github.com/fschulze/pytest-flakes/issues/26

**Comment by asmeurer** [comment]
Fixed by #29.
